parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetRecordReader.java

```diff
@@ -198,6 +206,10 @@ private void initializeInternalReader(ParquetInputSplit split, Configuration con
     }
 
     MessageType fileSchema = footer.getFileMetaData().getSchema();
+    Map<String, String> fileMetaData = footer.getFileMetaData().getKeyValueMetaData();
+    readContext = readSupport.init(new InitContext(configuration, InternalParquetRecordReader
+        .toSetMultiMap(fileMetaData), fileSchema));
```

Comment:
I don't think this is needed and should not be duplicated between this class and the `InternalParquetRecordReader` class. It is used here to create new MessageType objects that are passed into `VectorizedRecordReaderImplementation#readVectors` but I think we need to change that API, probably using `ColumnDescriptor[]` instead of `MessageType`.